src/util/spanparsing.h

```diff
@@ -36,21 +37,21 @@ bool Func(const std::string& str, Span<const char>& sp);
  */
 Span<const char> Expr(Span<const char>& sp);
 
-/** Split a string on every instance of sep, returning a vector.
+/** Split a string on any char found in separators, returning a vector.
  *
  * If sep does not occur in sp, a singleton with the entirety of sp is returned.
  *
  * Note that this function does not care about braces, so splitting
  * "foo(bar(1),2),3) on ',' will return {"foo(bar(1)", "2)", "3)"}.
  */
 template <typename T = Span<const char>>
-std::vector<T> Split(const Span<const char>& sp, char sep)
+std::vector<T> Split(const Span<const char>& sp, std::string_view separators)
 {
     std::vector<T> ret;
     auto it = sp.begin();
     auto start = it;
     while (it != sp.end()) {
-        if (*it == sep) {
+        if (separators.find(*it) != std::string::npos) {
```

Comment:
This is doing full scan of the iterators for each input character, thus the overall complexity is `O(input_length * number_of_separators)`. It can be optimized to `O(input_length + number_of_separators)` if a map of the separators is built beforehand that would allow checking whether a given character is a separator in a constant time.

This is mostly theoretical because it would matter if lots of separators are used and currently we only use a few. Anyway, out of curiosity, I tried that and compared the performance - in the current variant it is 734 ns/op while with the map it is 130 ns/op for 10 separators.

<details>
<summary>optimize Split()</summary>

```diff
 template <typename T = Span<const char>>
 std::vector<T> Split(const Span<const char>& sp, std::string_view separators)
 {
+    std::bitset<256> m;
+    for (const auto& sep : separators) {
+        m[static_cast<unsigned char>(sep)] = true;
+    }
     std::vector<T> ret;
     auto it = sp.begin();
     auto start = it;
     while (it != sp.end()) {
-        if (separators.find(*it) != std::string::npos) {
+        if (m[static_cast<unsigned char>(*it)]) {
             ret.emplace_back(start, it);
             start = it + 1;
         }
         ++it;
     }
     ret.emplace_back(start, it);
```
</details>

<details>
<summary>benchmark</summary>

```cpp
static void Split(benchmark::Bench& bench)
{
    bench.run([&] {
        (void)spanparsing::Split(
            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
            "bcdefghijk"
        );
    });
}

BENCHMARK(Split);
```
</details>

Feel free to ignore.